### ISSUE TITLE
Integration with existing CLI systems?

### ISSUE THREAD

**Issue by milesj** [terkelg/prompts#93]
I've been building a tool called [Boost](https://github.com/milesj/boost/) that provides an easy way to create CLI based dev tools. It's feature packed, but the one critical feature that's missing is a prompts system. The reason prompts don't exist is that the CLI in Boost completely hijacks `stdout` and `stderr`, by buffering all output, clearing lines/output automatically, and continuously re-rendering itself in the background (think Jest styled output). You can see this here: https://github.com/milesj/boost/blob/master/packages/core/src/Console.ts#L340

I've attempted to integrate this package and `inquirer` here: https://github.com/milesj/boost/pull/5. The major issue is both libraries operate on `stdout`, like clearing previous lines, which completely destroy Boost's CLI layer. Basically things like [this](https://github.com/terkelg/prompts/blob/master/lib/util/clear.js) and possibly [this](https://github.com/terkelg/prompts/blob/master/lib/elements/confirm.js#L63) are problematic. `stdout` can already be intercepted with the `out`, so the current functionality for this works pretty well.

I'd really like to use an existing open source prompts package instead of writing it myself, but I'm looking for the following support. **I'd be willing to submit a PR for these if you're open to them.*

1) All console operations can be intercepted with handlers passed to `prompts`. When not defined, they fallback to the existing implementation. For example:

```js
await prompts({
  onClear(lineCount) {},
  onMoveCursor() {}, // Maybe?
});
```

2) Customizing the colors used in the output. Boost has a [theme layer](https://github.com/milesj/boost/blob/master/packages/core/src/themes.ts) (support for color blindness) that is used when a terminal supports 255+ colors, otherwise it defaults to the standard ANSI colors. Something like the following would suffice.

```js
await prompts({
  onColor(message, color) {},
});
```

3) Localization is the last big piece. Many of the messages are hardcoded, like [this](https://github.com/terkelg/prompts/blob/f89f2a2a53b7995849ba3c225352f0b3c42f9976/lib/elements/text.js#L25). It would be great if we can customize these!

```js
await prompts({
  messages: {
    errorInvalidValue: 'Please enter a valid value!',
  },
});
```

Looking forward to hearing your thoughts. If you're cool with this, I can go ahead and start contributing.

**Comment by terkelg** [comment]
Thanks @milesj. If we can implement this in simple way. I'm not sure the 3rd one is needed since all prompts already provide a error message option.

I think Prompts need a better and more coherent clearing/rendering system. Maybe this is a good opportunity to implement that.

**Comment by milesj** [comment]
@terkelg Good point! Thanks for taking the time to read this. I'll prototype a few things, as some of these requests might not be needed, or can be very basic. I'll report back once I've dug in further.

**Comment by milesj** [comment]
Oh also, since I use TypeScript, I could help audit/write them while I'm looking into this.
